Chaincase.UI/ViewModels/CoinJoinViewModel.cs

```diff
@@ -0,0 +1,391 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+using System.Security;
+using Chaincase;
+using Chaincase.Common;
+using Chaincase.Common.Contracts;
+using Chaincase.Common.Models;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using ReactiveUI;
+using Splat;
+using WalletWasabi.Blockchain.Keys;
+using WalletWasabi.CoinJoin.Client.Clients.Queuing;
+using WalletWasabi.CoinJoin.Client.Rounds;
+using WalletWasabi.CoinJoin.Common.Models;
+using WalletWasabi.Helpers;
+using WalletWasabi.Logging;
+
+namespace Chaincase.UI.ViewModels
+{
+	public class CoinJoinViewModel : ReactiveObject
+	{
+		protected Global Global { get; }
+        private CompositeDisposable Disposables { get; set; }
+
+        private INotificationManager notificationManager;
+
+        private string _coordinatorFeePercent;
+		private int _peersRegistered;
+		private int _peersNeeded;
+
+		private RoundPhaseState _roundPhaseState;
+		private DateTimeOffset _roundTimesout;
+		private TimeSpan _timeLeftTillRoundTimeout;
+		private Money _requiredBTC;
+		private Money _amountQueued;
+		private bool _isDequeueBusy;
+		private bool _isEnqueueBusy;
+        private bool _isQueuedToCoinJoin = false;
+		private string _balance;
+        private string _toastErrorMessage;
+        private bool _shouldShowErrorToast;
+        private SelectCoinsViewModel _selectCoinsViewModel;
+        protected StatusViewModel _statusViewModel;
+
+        public CoinJoinViewModel(Global global) 
+        {
+            Global = global;
+            Global.NotificationManager.RequestAuthorization();
+            CoinList = new SelectCoinsViewModel(global);
+            _statusViewModel = new StatusViewModel(global); 
+
+            if (Disposables != null)
+            {
+                throw new Exception("Wallet opened before it was closed.");
+            }
+
+            Disposables = new CompositeDisposable();
+			// Sum UTXOs
+			//SetBalance();
+			// Start with 0 btc until user chooses some UTXOs
+			AmountQueued = Money.Zero;
+            // Infer coordinator fee
+            var registrableRound = Global.Wallet.ChaumianClient.State.GetRegistrableRoundOrDefault();
+            CoordinatorFeePercent = registrableRound?.State?.CoordinatorFeePercent.ToString() ?? "0.003";
+
+            // Select most advanced coin join round
```

Comment:
does this comment provide information that could not otherwise be inferred?

Reply:
It makes it easier to skim the file and look for exactly what I am looking for without needing to read individual vars. Also, it's just a comment.